remove redundant functions/nested calls/tests

pine3ree opened the pull request:
The UrlExtension has basically duplicated UrlHelper and ServerUrlHelper methods proxying to them.
Plates calls to the registered function 'url' result in calling:

UrlExtension::generateUrl => UrlHelper::generate => UrlHelper::__invoke

Registering the invokable helpers directly (when possible) removed 2 redundant function calls :

Plates calls for 'url' result, for instance, in calling just UrlHelper::__invoke

The EscaperExtension uses the escaper's methods directly as well.

pine3ree committed:
remove redundant functions/nested calls/tests

weierophinney commented on test/Extension/UrlExtensionTest.php:
```diff
@@ -43,94 +43,15 @@ public function testRegistersUrlFunctionWithEngine()
         $engine = $this->prophesize(Engine::class);
         $engine
-            ->registerFunction('url', [$this->extension, 'generateUrl'])
+            ->registerFunction('url', $this->urlHelper)
```
I think we did it this way originally as the original versions of Plates we targeted didn't understand functors (i.e., you couldn't pass just an instance, but also needed to pass the function to invoke). Additionally, the rationale for the `generate()` method was because the notation when composing a functor as a property of another class makes invoking it awkward (e.g., `($this->urlHelper)(...)` vs `$this->urlHelper->generate(...)`). But I absolutely agree that, since this works, this is a good change to make here.

pine3ree replied:
If I remember well Plates also performs further tests when it encounters a callable array expression...

pine3ree replied:
if a callable array is still preferred, I would use [$this->urlHelper, '__invoke'], it's less "elegant" but still skips one internal proxy call...

weierophinney replied:
No, I like your solution here. Fewer layers of indirection. I was just noting why it was the way it was originally, in case folks come here wondering why it changed.

pine3ree replied:
@weierophinney Can I just change the target develop branch just with the "edit" button here on gh web interface? I remember doing it once and having issues....thanks!
(update)
It should not cause issue, I see just the one usual extra commit in master (minore version bump etc in readme file)

michalbundyra replied:
@pine3ree Yes, you can change the target on github but you also need rebase your changes and push force.

pine3ree replied:
@webimpress rebased to develop and targeted to develop

pine3ree replied:
@webimpress there are 3 or 4 null-coalescing replacements to be made here and there and maybe we could add a comment to remind the removal of the redundant test cases in v. 3.0

michalbundyra replied:
@pine3ree I guess you can add comments on these tests.

null-coalescing replacements you can do as it is not gonna affect anything, right?

pine3ree replied:
null-coalescing: it was just a reminder-comment for myself, since it's unrelated it's better to use a different PR for those minor changes. :-) kind regards

michalbundyra replied:
yeah, can submit separate PR to master - as improvement, no functionality change.

weierophinney commented on src/Extension/UrlExtension.php:
```diff
@@ -47,43 +47,8 @@ public function __construct(UrlHelper $urlHelper, ServerUrlHelper $serverUrlHelp
-    /**
-     * Generate a fully qualified URI, relative to $path.
-     */
-    public function generateServerUrl(string $path = null) : string
```
This is a BC break, since it removes public methods, even if they're not used internally.

Instead, please keep them, but add the notation `@deprecated since 2.2.0; to be removed in 3.0.0. These methods were used originally internally when registering functions with the Plates engine; we now register the (URL|Server URL) helper instance directly.`. 

This also means you'll need to rebase this against the `develop` branch and update the pull request to indicate it's against that branch.

pine3ree replied:
right, they were marked "public"...

weierophinney reviewed (changes requested):
:+1: , but it needs to be updated to deprecate the old methods instead of remove them, which in turn means it needs to be rebased and pushed against the develop branch.

pine3ree committed:
add back the removed redundant function (and related tests) to avoid BC
break, since they had public visibility, but mark them as deprecated
and scheduled to be removed in next major version.

weierophinney committed:
docs: adds CHANGELOG entries for #37

Also removes empty stub for 2.1.1 release, and sets date for 2.2.0
release.

weierophinney commented:
Thanks, @pine3ree!
